source/agora/network/DNSResolver.d

```diff
@@ -86,10 +86,16 @@ public final class VibeDNSResolver : DNSResolver
 
         Params:
             msg = DNS message
+            tcp = Query the server with TCP, default is UDP
 
     ***************************************************************************/
 
-    public override ResourceRecord[] query (Message msg) @trusted
+    public override ResourceRecord[] query (Message msg, bool tcp = false) @trusted
```

Comment:
If we add one more level of indirection, I think we should make `query` `final` and have `queryUDP` / `queryTCP` be the points of override. Otherwise `query` becomes a [leaky abstraction](http://www.joelonsoftware.com/articles/LeakyAbstractions.html).